cinema/views.py

```diff
@@ -40,6 +48,91 @@ def movie_detail(request, pk):
 
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
+    if request.method == "PATCH":
+        serializer = MovieSerializer(movie, data=request.data, partial=True)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_200_OK)
+
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
     if request.method == "DELETE":
         movie.delete()
         return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class GenreList(APIView):
+
+    def get(self, request):
+        genres = Genre.objects.all()[:20]
```

Comment:
Slicing is not the best idea here. The problem of getting too much data can be solved using pagination, which we will soon encounter. As for now, I'd suggest leaving it as is, i.e. `Genre.objects.all()`.